src/Persistence/IteratorQuery.php

```diff
@@ -311,7 +313,7 @@ protected function evaluateIf($v1, $operator, $v2): bool
     {
         switch (strtoupper((string) $operator)) {
             case Condition::OPERATOR_EQUALS:
-                $result = is_array($v2) ? $this->evaluateIf($v1, Condition::OPERATOR_IN, $v2) : $v1 == $v2;
+                $result = is_array($v2) ? $this->evaluateIf($v1, Condition::OPERATOR_IN, $v2) : $v1 === $v2;
```

Comment:
very good, all repos except this uses strict types already, what is left to use strict types also for this repo?

Reply:
One of the remaining points in this repo is to streamline the Field typecasting. It has to do with the strict types.